### PR TITLE
Make examples run on localhost

### PR DESCRIPTION
**Pull request by birkskyum**
As a contributor, it's quite useful to be able to run and tweak the examples on localhost. 

Currently, almost no example will run out of the box, yielding just a white canvas and a terminal with errors. Most require contributors to apply a personal credit card in signing up for various third-parties, and/or modifying each example to open assets which can be rather tricky (requires finding appropriate style,  browsing maputnik to find layer names, and setting up loading for maki icons etc.  -  basically the content of this PR)

To resolve these issues, I've found ways to showcase the examples with open resources, in particular the demotiles and the free carto voyager style (and for 1 hybrid sat example, the arcgis orto). I've ported the examples where the free styles are perfectly capable of demonstrating the respective feature, and left the rest untouched.

 - [X] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!

### CODE REVIEW

**Comment by HarelM** [comment]
Can you clarify the problem? 
most examples that use third party tile providers work well with local host and you don't need a credit card to make them run using localhost, as far as I know...

File: test/examples/variable-offset-label-placement.html [hunk 1]
```diff
@@ -101,11 +101,26 @@
...
     const map = new maplibregl.Map({
         container: 'map',
-        style: 'https://api.maptiler.com/maps/streets/style.json?key=get_your_own_OpIi9ZULNHzrESv6T2vL',
+        style: 'https://basemaps.cartocdn.com/gl/voyager-gl-style/style.json',
         center: [-77.04, 38.907],
         zoom: 11.15
     });
 
+    const requestedIcons = {};
+    map.on('styleimagemissing', (e) => {
```
Comment (HarelM) [inline test/examples/variable-offset-label-placement.html:110]: Is there a better way to solve this? This mainly clutters the example with code that is not "important".
Remember that these examples are shown in the doc pages on maplibre.org too and are not only for local testing while developing.

File: test/examples/popup-on-click.html [hunk 2]
```diff
@@ -145,17 +165,25 @@
...
-                'icon-image': '{icon}_15',
-                'icon-overlap': 'always'
+                'icon-image': '{icon}-15',
+                'icon-overlap': 'always',
+                'icon-size': 1.2
             }
         });
 
+        // Create a popup, but don't add it to the map yet.
+        const popup = new maplibregl.Popup({
```
Comment (HarelM) [inline test/examples/popup-on-click.html:182]: Where is this popup used?

File: test/examples/map-tiles.html [hunk 3]
```diff
@@ -18,27 +18,21 @@
     const map = new maplibregl.Map({
         container: 'map', // container id
         style: {
-            'version': 8,
-            'sources': {
-                'raster-tiles': {
-                    'type': 'raster',
-                    'tiles': [
-                        // NOTE: Layers from Stadia Maps do not require an API key for localhost development or most production
```
Comment (HarelM) [inline test/examples/map-tiles.html:26]: This is what I was referring to.

**Comment by HarelM** [comment]
What's that license of 
https://basemaps.cartocdn.com/gl/voyager-gl-style/style.json
Can we actually use it "for free"?

File: test/examples/map-tiles.html (hunk 3 above)
Comment (birkskyum) [reply test/examples/map-tiles.html:26]: I don't know. The comment says that it should work on localhost, by my browser console told a different story:
 
Screenshot 2024-09-07 at 20 36 50

File: test/examples/popup-on-click.html (hunk 2 above)
Comment (birkskyum) [reply test/examples/popup-on-click.html]: it's not - i've removed it

**Comment by birkskyum** [comment]
> What's that license of https://basemaps.cartocdn.com/gl/voyager-gl-style/style.json Can we actually use it "for free"?

It's a BSD-3 clause https://github.com/CartoDB/basemap-styles/blob/master/LICENSE.md

**Comment by birkskyum** [comment]
> Can you clarify the problem? most examples that use third party tile providers work well with local host and you don't need a credit card to make them run using localhost, as far as I know...

The problem is that opening almost all examples in the dev server looks like this unless manually modified:
Screenshot 2024-09-07 at 20 41 54

File: test/examples/map-tiles.html (hunk 3 above)
Comment (HarelM) [reply test/examples/map-tiles.html:26]: Is it possible that you are going to 0.0.0.0 instead of localhost?

Comment (birkskyum) [reply test/examples/map-tiles.html:26]: omg... you're right. This has been bugging me forever. The reason i go 0.0.0.0 is this:

Screenshot 2024-09-07 at 20 53 10

Comment (birkskyum) [reply test/examples/map-tiles.html:26]: Made a PR #4675  to fix that. It appear to be solving my problem

**Pull request closed without merging**